src/DotPulsar/ConsumerOptions.cs

```diff
@@ -66,19 +68,33 @@ public sealed class ConsumerOptions
         /// </summary>
         public static readonly TimeSpan DefaultNegativeAcknowledgeRedeliveryDelay = TimeSpan.FromMinutes(1);
 
+        /// <summary>
+        /// Default mode is to load one message from partition at a time and then move to the next partition.
+        /// </summary>
+        public static readonly bool DefaultOneMessageFromPartitionAtATime = true;
+
+        /// <summary>
+        /// Default number of messages to load from one partition before moving to the next one.
+        /// </summary>
+        public static readonly uint DefaultNumberOfMessagesToReadFromPartition = 10;
```

Comment:
I do think the default should remain 1 as that is the expected behavior and this is a new feature custom added to only our client.  In our microbatching use case, it makes sense to force us to specify we want otherwise.

Reply:
fixed